### ISSUE TITLE
[DRAFT][Bug] Notice of Disagreement - Issues for review opt-in page displays even without old issues

### ISSUE THREAD

**Issue by tlei123** [department-of-veterans-affairs/va.gov-team#24993]
[DRAFT: Will finalize ticket details once all QA test-runs are completed.]

## What happened?
On Staging **[Request a Board Appeal form](https://staging.va.gov/decision-reviews/board-appeal/request-board-appeal-form-10182/introduction)**, **Issues for review** [opt-in] page displayed despite having no old issues selected/added [with decision date before 2/19/2019].

## Specs:
- Test User _(if applicable)_: [use any of the following]
  - `vetsgov+test239@id.me` (Medge)
  - `vetsgov+test233@id.me` (Cara)
  - `vetsgov+test238@id.me` (Nerea)

## Steps to Reproduce
- URL: https://staging.va.gov/decision-reviews/board-appeal/request-board-appeal-form-10182/introduction
- Test User _(if applicable)_: [see **Specs** above]
- Start a new request to ensure no Save-In-Progress prefills occur.
- Advance to **Issues for Review** [select] page, completing any required fields and clicking **Continue** on the way.
- Select a **recent issue** [with Decision date **after** 2/19/2019], then **Continue**.
- On **Issues for review** [areas of disagreement] page, choose any **option**, then **Continue**.
- Observe that **Issues for review** [opt-in] page is displayed, despite having no old issue selected [w/ Decision date before 2/19/2019].

**Screenshot**:
![nod-bug-opt-in-not-skipped](https://user-images.githubusercontent.com/587583/118919155-dcb98000-b8e8-11eb-871a-239714ab6d6e.png)

## Desired behavior
- [ ] With no old issues selected/added, **Issues for review** [opt-in] page should be skipped.

## Acceptance Criteria
[see **Desired behavior** above]

## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to Epic** (what body of work is this a part of? possibly `Ongoing Maintenance`)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `triage`, `tools-improvements`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [X] **Labeled with `Bug`**

**Comment by Mottie** [comment]
Related ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/25070

**Comment by Mottie** [comment]
The list of issues has been removed. We'll restore this list once Lighthouse provides a method to indicate legacy issues.

**Comment by tlei123** [comment]
@Mottie , I'm thinking we should roll this Bug back from Validate to Icebox or Backlog, so that your follow-up fix (distinguish legacy issues from new ones to manage Opt-in page show/hide) doesn't drop off our radar when Lighthouse does provide what you need.

I've validated your interim fix (hiding all issues from page) successfully on Staging.

**Comment by Mottie** [comment]
> roll this Bug back from Validate to Icebox or Backlog

Yes please!

**Comment by vhenry07** [comment]
Awaiting Lighthouse endpoint completion

**Comment by Mottie** [comment]
We were planning on working on this ticket in this sprint, but I discovered that the legacy appeals API data was added to the Higher-Level Review endpoint, but not yet added to the Notice of Disagreement endpoint.

**Comment by Mottie** [comment]
Work for this has been merged into staging... if a user has _any_ legacy appeals, or have manually added an issue, this opt-in page will appear. The list of issues will not be included on the page.

User 233 (`vetsgov+test233@id.me`) does not have any legacy issues, and user 236 (`vetsgov+test236@id.me`) has 6 issues, but you wouldn't know the amount from looking at the pages.

Related: https://github.com/department-of-veterans-affairs/va.gov-team/issues/35366

**Comment by tlei123** [comment]
> User 233 (`vetsgov+test233@id.me`) does not have any legacy issues, and user 236 (`vetsgov+test236@id.me`) has 6 issues, but you wouldn't know the amount from looking at the pages.

@Mottie , did you mean all legacy issues get hidden from the initial **Issues for Review** [list] page?

I just checked Staging.  233's not testable up there due to some kinda profile issue.  236 displayed only non-legacy issues [decision dates all March 15, 2021], but Opt-in page still displayed [even though I didn't manually Add one].

Is #35366 implemented in a way that legacy issues automatically triggers the Opt-in page regardless of whether they're shown/selected?

Pls. let me know.  Thx much!

**Comment by Mottie** [comment]
> all legacy issues get hidden from the initial **Issues for Review** [list] page?

Yes, we can't correlate the rated disabilities with legacy issues, only a combined name and description is provided and there is no easy way to match it to a contestable issue

> implemented in a way that legacy issues [from incoming data] automatically trigger the Opt-in page regardless of whether they're shown/selected?

Yes, pretty much. If the user chooses any issue, we can't assume it is or isn't a legacy issue. And if they choose none, and add one, we have no way of knowing if it is legacy or not

> 233's not testable up there due to some kinda profile issue

Hmm, are you using one of these users? https://dsva.slack.com/archives/C5AGLBNRK/p1601297745026400?thread_ts=1601069567.024300&cid=C5AGLBNRK - HLR/NOD mock users aren't included on the main list of users

**Comment by tlei123** [comment]
Thx, @Mottie !  Just checked again and test233's profile issue's gone today.

Yes,  I'm using ...test233 & ...test236.

**Comment by tlei123** [comment]
Sorry, @Mottie !  test233 (Cara) is still showing Opt-in page even w/o manually Adding an issue.  See video below.

https://user-images.githubusercontent.com/587583/150572340-c90d6d44-8023-4de4-b8f4-e672e3673b6b.mp4

**Comment by Mottie** [comment]
Hmm, you're looking at HLR v1 somehow? I set the feature flag of all staging users to show v2, which doesn't have the blue border:

![HLR v2 contestable issues page without a blue border](https://user-images.githubusercontent.com/136959/150574529-5af6e69e-8e55-45d3-9973-55dc3a018f6b.png)

**Comment by Mottie** [comment]
Oh, you're looking at NOD (form 10182)... sorry, I was thinking this ticket was for HLR. My mistake.

**Comment by Mottie** [comment]
@tlei123 This ticket is now obsolete. We have recently removed the opt-in checkbox, to near the beginning of the flow, and the changed the content - see https://github.com/department-of-veterans-affairs/va.gov-team/issues/36953

Filing deadline page (formerly opt-in page)

![Filing deadline page, now with less checkboxes](https://user-images.githubusercontent.com/136959/154330467-111f3b28-bcf8-4506-9eeb-1fa2a8148ae2.png)

**Comment by tlei123** [comment]
Closing as outdated.
